index.js

```diff
@@ -1,20 +1,252 @@
-const readline = require('readline');
-const readlineInterface = readline.createInterface(process.stdin, process.stdout);
+const { read } = require("fs");
+const readline = require("readline");
+const readlineInterface = readline.createInterface(
+  process.stdin,
+  process.stdout
+);
 
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     readlineInterface.question(questionText, resolve);
   });
 }
 
-start();
+//a player inventory that can be added to, viewed and removed from
+let playerInventory = [];
+//the player begins at home
+let currentLocation = "home";
+
+// Location class constructor
+class Location {
+  constructor(locName, locDescription, locInventory, unlocked, possibleLoc) {
+    this.locName = locName;
+    this.locDescription = locDescription;
+    this.locInventory = locInventory;
+    this.unlocked = unlocked || false;
+    this.possibleLoc = possibleLoc;
+  }
+  //move seems to not be cooperating with the item inventory funcionality
+  move(newLoc) {
+    if (this.possibleLoc.includes(newLoc)) {
+      currentLocation = newLoc;
+      return `You are now in the ${currentLocation}. It's ${locLookup[newLoc].locDescription}`;
+    } else {
+      return `You can't go directly from ${currentLocation} to ${newLoc} `;
+    }
+  }
+
+  addInv(item) {
+    locLookup[currentLocation].locInventory.push(item);
+    return `You drop the ${item} on the ground.`;
+  }
+}
+// Item class constructor, mostly working as intended
+class Item {
+  constructor(name, description, action, takeable, readable) {
+    this.name = name;
+    this.description = description;
+    this.action = action;
+    this.takeable = takeable || false;
+    this.readable = readable || false;
+  }
+  //action functions
+  take() {
+    if (this.takeable) {
+      playerInventory.push(this.name);
+      return `You picked up ${this.name}`;
+    } else {
+      return "You can't take that";
+    }
+  }
+
+  use() {
+    if (this.name === "note") {
+      return this.action + this.description;
+    } else {
+      return this.action;
+    }
+  }
+
+  read() {
+    if (this.readable) {
+      return this.readable;
+    } else {
+      return `You can't read a ${this.name}`;
+    }
+  }
+  // drop function not working yet
+  drop() {
+    if (this.name) {
+      const dropped = playerInventory.indexOf(this.name);
+      playerInventory.splice(dropped, 1);
+      let dropInv = new Location();
+      return dropInv.addInv(this.name);
+    }
+  }
```

Comment:
These need to be on the global scope so that they triggered within your async function